File: c/src/common/honolulufishchowder.c

#include "honolulufishchowder.h"


soup_t * honolulufishchowder_new()
{
	soup_t * soup = soup_new();
	soup->name = "OpakapakaFishChowder";
	arraylist_string_add( soup->ingredients,"1 Pound Fresh Opakapaka Fish");
	arraylist_string_add( soup->ingredients,"1 cup pineapple chunks");
	arraylist_string_add( soup->ingredients,"1/2 cup coconut milk");
	arraylist_string_add( soup->ingredients,"1/4 cup SPAM");
	arraylist_string_add( soup->ingredients,"1/4 cup taro chips");
   return soup;
}

File: clojure/src/common/honolulufishchowder.clj
(defclass honolulufishchowder
   (fishchowder)
   ())

(defmethod initialize-instance :after ((this honolulufishchowder) &key)
   (setf (slot-value this 'name) "OpakapakaFishChowder")
   (setf (slot-value this 'ingredients) (list
        "1 Pound Fresh Opakapaka Fish"
        "1 cup pineapple chunks"
        "1/2 cup coconut milk"
        "1/4 cup SPAM"
        "1/4 cup taro chips"
	)))
